conversions/hexadecimal_to_octal2.c

```diff
@@ -0,0 +1,116 @@
+/**
+ * @file
+ * @brief Convert hexadecimal number to octal number
+ * @details
+ * The input is valid from 0 to 0xFFFF_FFFF_FFFF_FFFF.
+ *
+ * At first, this program converts a hex string to an unsigned long long
+ * decimal, and then to an octal string.
+ *
+ * When there is an invalid character in input string, this program stops
+ * parsing and converts the string until that character.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+/**
+ * Convert a hexadecimal number to octal number.
+ * @param hex Hexadecimal number to convert.
+ * @returns A pointer to the converted octal string.
+ */
+char *hex_to_oct(const char *hex)
+{
+#define MAX_OCT_STR_LEN 23 /* 17_7777_7777_7777_7777_7777 */
+    char *octal = (char *)malloc(sizeof(char) * MAX_OCT_STR_LEN);
+    unsigned long long decimal = 0;
+    int i = 0;
+    int len;
+
+    if (hex == NULL)
+    {
+        octal[0] = '\0';
+        return octal;
+    }
+
+    /* Hexadecimal to decimal conversion */
+    while (*hex != '\n' && *hex != '\0')
+    {
+        char ch = *hex;
+
+        if (ch >= '0' && ch <= '9')
+        {
+            ch -= '0';
+        }
+        else if (ch >= 'a' && ch <= 'f')
+        {
+            ch = ch - 'a' + 10;
+        }
+        else if (ch >= 'A' && ch <= 'F')
+        {
+            ch = ch - 'A' + 10;
+        }
+        else
+        {
+            printf("Invalid hexadecimal input: %c\n", ch);
+            break;
+        }
+
+        decimal *= 16;
+        decimal += ch;
+        hex++;
+    }
+
+    /* Decimal to octal conversion */
+    if (decimal == 0)
+    {
+        octal[0] = '0';
+        len = 1;
+    }
+    else
+    {
+        i = 0;
+        while (decimal > 0)
+        {
+            octal[i] = '0' + decimal % 8;
+            i++;
+            decimal /= 8;
+        }
+
+        len = i;
+    }
+
+    octal[len] = '\0';
```

Comment:
the use of memset above, makes this line redundant. no harm in keeping it though. Just an FYI. 
Some inline comments would be helpful for readers.